test/blackbox-tests/deriving-abstract-deprecation.t

```diff
@@ -0,0 +1,56 @@
+Deprecations alerts for deriving abstract
+
+  $ . ./setup.sh
+
+  $ cat > dune-project <<EOF
+  > (lang dune 3.8)
+  > (using melange 0.1)
+  > EOF
+
+  $ cat > dune <<EOF
+  > (library
+  >  (modes melange)
+  >  (name x)
+  >  (preprocess
+  >   (pps melange.ppx)))
+  > EOF
+
+Shows when using the creation function
+
+  $ cat > x.ml <<EOF
+  > type chartDataItemType = { height : int; foo : string } [@@deriving abstract]
+  > 
+  > let t = chartDataItemType ~height:2 ~foo:"bar"
+  > EOF
+
+  $ dune build ./.x.objs/melange/x.cmj
+  File "x.ml", line 3, characters 8-25:
+  3 | let t = chartDataItemType ~height:2 ~foo:"bar"
+              ^^^^^^^^^^^^^^^^^
+  Error (alert deprecated): chartDataItemType
+  The `@deriving abstract` deriver is deprecated, use `@deriving dynamicKeys` instead.
+  [1]
+
+Shows when using setters or accessors
+
+  $ cat > x.ml <<EOF
+  > type chartDataItemType = { height : int; foo : string } [@@deriving abstract]
+  > 
+  > let t = chartDataItemType ~height:2 ~foo:"bar"
+  > let u = heightGet t
+  > EOF
+
+  $ dune build ./.x.objs/melange/x.cmj
+  File "x.ml", line 3, characters 8-25:
+  3 | let t = chartDataItemType ~height:2 ~foo:"bar"
```

Comment:
this location will be a lot better if you use the location on the attribute.

Reply:
I am not sure I understand your suggestion. The deriver has no access to the original ast (which is where the attribute is), it can only insert the deprecation alert on the newly created ast nodes. I believe it's the same root cause that prevented making the type abstract originally when the deriver was introduced.

More generically, it feels that implementing `deriving abstract` with a pure deriver was a too limiting choice for the use case.

Reply:
I don't know what I was thinking, tbh. You're right, of course.